### ISSUE TITLE
Bitcoin-core Policy following... policy

### ISSUE THREAD

**Issue by dcousens** [bitcoinjs/bitcoinjs-lib#640]
I think it is safe to say our "policy" for node policy related code is the following:
- ~~`bitcoinjs.script.*`~~ `bitcoinjs.template.*` functions should follow the script "templates" typically used in the community (and consequently, the bitcoin blockchain).
  This is a matter of community consensus,  but is probably going to always be closely related to what bitcoin core does by default.
  They are quite plainly just pattern matching functions, and should be treated as such.

`bitcoinjs.script.decompile` is consensus bound only,  it does not reject based on policy.
`bitcoinjs.script.compile` will adhere to bitcoin-core `IsStandard` policies rules such as minimalpush in https://github.com/bitcoinjs/bitcoinjs-lib/pull/638.

All ECDSA `sign` operations will adhere to `IsStandard` policies such as `LOW_S`, but `verify` should not reject them.

For rejecting non-standard `decoding`, you should use external modules to this library (which we could probably build a list of, and may even be in this organization).

**TLDR:** 
Where "standards compliant" refers to the default policies of bitcoin-core,  we adhere to the following:
- Any "creation" event must create standards-compliant data
- Any "validation" event must allow for non-standards compliant data (aka, consensus only)

For stricter validation,  use an external module which we [might have] provided.
We will use these modules to test `bitcoinjs-lib` against to ensure standards-compliant "encoding".

**Comment by dcousens** [comment]
@junderw I think we can move this and the contributing issue to a CONTRIBUTING.md?

**Comment by dcousens** [comment]
See https://github.com/bitcoinjs/bitcoinjs-lib/pull/1042
